src/input/gamepad/gamepadCalibration.js

```diff
@@ -0,0 +1,232 @@
+import type {Gamepad, Button} from "./gamepad";
+
+function scanForButton ( buttons0 : Array<Button>, buttons1 : Array<Button>
+                       , axes0 : Array<number>, axes1 : Array<number>
+                       ) : null | ButtonInfo {
+  let buttonInfo = null;
+
+  const bLg = buttons1.length;
+
+  for (let i = 0; i < bLg; i++) {
+    if (detectedButtonPressed(buttons0[i].pressed, buttons1[i].pressed)) {
+      buttonInfo = { kind : "pressed", index : i };
+      break;
+    }
+    else if (detectedButtonValue(buttons0[i].value, buttons1[i].value)) {
+      buttonInfo = { kind : "value", index : i, threshold : 0.75 };
+      break;
+    }
+  }
+  if (buttonInfo === null) {
+    const aLg = axes1.length;
+    for (let j = 0; j < aLg; j++) {
+      if (detectedButtonValue(axes0[j], axes1[j])) {
+        buttonInfo = { kind : "axis", index : j, threshold : 0.75 };
+      }
+    }
+  }
+  return buttonInfo;
+};
+
+function detectedButtonPressed ( pressed0 : bool, pressed1 : bool ) : bool {
+  return pressed1 && !pressed0;
+};
+
+function detectedButtonValue ( value0 : number, value1 : number ) : bool {
+  return (value0 < 0.25 && value1 > 0.75);
+};
+
+
+function scanForTrigger ( buttons0 : Array<Button>, buttons1 : Array<Button>
+                        , axes0 : Array<number>, axes1 : Array<number>
+                        ) : null | TriggerInfo {
+  const aLg = axes1.length;
+  let minMax;
+  let triggerInfo = null;
+  for (let i = 0; i < aLg && triggerInfo === null; i++) {
+    minMax = detectedTrigger( axes0[i], axes1[i]);
+    if (minMax !== null) {
+      triggerInfo = { kind : "axis", index : i, min : minMax[0], max : minMax[1] };
+    }
+  }
+  if (triggerInfo === null) {
+    const bLg = buttons1.length;
+    for (let j = 0; j < bLg && triggerInfo === null; j++) {
+      minMax = detectedTrigger(buttons0[j].value, buttons1[j].value);
+      if (minMax !== null) {
+        triggerInfo = { kind : "value", index : j, min : minMax[0], max : minMax[1] };
+      }
+    }
+  }
+  return triggerInfo;
+}
+
+function detectedTrigger ( axis0 : number, axis1 : number ) : null | [number, number] {
+  if (Math.abs(axis1 - axis0) < 0.5) {
+    return null;
+  }
+  else {
+    return getMinAndMax(axis0, axis1);
+  }
+}
+
+function getMinAndMax(axis0 : number, axis1 : number) : [number, number] {
+  const min = axis0 < -0.87 ? -1 : axis0 < -0.5 ? -0.866 : axis0 > 0.87 ? 1 : axis0 > 0.5 ? 0.8667 : 0;
+  const max = min === 0 ? Math.sign(axis1) : -min;
+  return [min, max];
+}
+
+function scanForStick ( buttons0 : Array<Button>
+                      , buttonsL : Array<Button>, buttonsR : Array<Button>
+                      , buttonsU : Array<Button>, buttonsD : Array<Button>
+                      , axes0 : Array<number>
+                      , axesL : Array<number>, axesR : Array<number>
+                      , axesU : Array<number>, axesD : Array<number> ) : null | StickInfo {
```

Comment:
For control sticks we use 5 snapshots: one at neutral, and one each for left/right/top/bottom control stick input.